mne/io/reference.py

```diff
@@ -298,6 +298,12 @@ def set_eeg_reference(inst, ref_channels=None, copy=True, verbose=None):
                 inst.info['custom_ref_applied'] = False
                 inst.add_proj(make_eeg_average_ref_proj(inst.info,
                               activate=False))
+                # Apply the reference if data has been preloaded
+                if hasattr(inst, '_data') or hasattr(inst, 'data'):
```

Comment:
otherwise it would become something like:

    if ((hasattr(inst, 'preload') and inst.preload) or hasattr(inst, 'data'))

not necessarily clearer.